python/triton/runtime/autotuner.py

```diff
@@ -136,24 +152,25 @@ def _bench(self, *args, config, **meta):
         # augment meta-parameters with tunable ones
         current = dict(meta, **config.all_kwargs())
         full_nargs = {**self.nargs, **current}
+        hook_args = self._get_hook_args(full_nargs)
 
         def kernel_call():
             if config.pre_hook:
                 config.pre_hook(full_nargs)
-            self.pre_hook(args)
+            self.pre_hook(hook_args)
```

Comment:
This would be a BC break, but it might be better to match `config.pre_hook` and just call the hooks with `full_nargs` directly. This should avoid the `triton.heuristics` problem entirely and just be more consistent. 

@Jokeren do you know if these hooks are widely used?

Reply:
Yeah, my assumption here was that we don't want to change the existing `args`-based API. However, I agree that it would be more convenient to operate on named args in the hooks (otherwise, it's tedious for a user to figure out what the positions are).

Checked internally: seems that folks are almost always using `config.pre_hook`, probably due to the mentioned reason. Obv, not sure what happens in the wild. If folks are fine, I can change the PR to make `Autotuner.{pre,post}_hook`s API-wise like the `Config.pre_hook`. Please let me know.

Reply:
`config.pre_hook` doesn't work for the profiler. It needs both an enter hook and an exit hook

Reply:
If we can keep `pre_hook` but just to adjust its arguments, I'm OK with it

Reply:
@Jokeren to make sure I understand correctly: you're saying you're ok to keep `Autotuner.pre_hook` and `Autotuner.post_hook`, but change the first parameter from list of args to dict of kwargs in both?

Reply:
Generally, yes, but there's probably an exception.

I think you still need to add `reset_only` for `self.pre_hook(hook_args, reset_only=True)`.

Reply:
Yep, `reset_only` and `exception` will stay as separate hook parameters. Just like in the previous hook signature, the kwargs (before this PR: args) of the kernel call will be passed as the first hook parameter: a dict (before this PR: list). I'll update the PR soon.